nnfabrik/templates/transfer/transferred_trained_model.py

```diff
@@ -0,0 +1,310 @@
+                        "seed",
+                        "collapsed_history",
+                        "data_transfer",
+                    )
+                    & transfers
+                    & post_restr  # restrict with post_rest
+                )
+
+                if key_source is not None:
+                    key_source = key_source + transfers.proj()
+                else:
+                    key_source = transfers.proj()
+
+        # normal entries as a combination of Dataset, Model, Trainer, and Seed tables
+        step_0 = self.model_table * self.dataset_table * self.trainer_table * self.seed_table
+        # add transfer_step, collapsed_history and data_transfer as prim keys
+        base = dj.U("transfer_step", "collapsed_history", "data_transfer",) * step_0.proj(
+            transfer_step="0",
+            collapsed_history='""',
+            data_transfer="0",
+        )
+
+        if key_source is None:
+            return base.proj()
+        else:
+            return key_source + base.proj()
+
+    def get_full_config(self, key=None, include_state_dict=True, include_trainer=True):
+        ret = super().get_full_config(
+            key=key,
+            include_state_dict=include_state_dict,
+            include_trainer=include_trainer,
+        )
+        prev_key = self.CollapsedHistory().get_prev_key(key)
+        # retrieve corresponding model state (and overwrite possibly retrieved state)
+        if include_state_dict and (self.ModelStorage & prev_key):
+            with tempfile.TemporaryDirectory() as temp_dir:
+                state_dict_path = (self.ModelStorage & prev_key).fetch1("model_state", download_path=temp_dir)
+                ret["state_dict"] = torch.load(state_dict_path)
+                ret["model_config"]["transfer"] = True
+        # retrieve data if present (if previous step did data transfer)
+        if self.DataStorage & prev_key:
+            with tempfile.TemporaryDirectory() as temp_dir:
+                data_path = (self.DataStorage & prev_key).fetch1("transfer_data", download_path=temp_dir)
+                ret["dataset_config"]["transfer_data"] = np.load(data_path)
+        return ret
+
+    def make(self, key):
+        """
+        Given key specifying configuration for dataloaders, model and trainer,
+        trains the model and saves the trained model.
+        """
+
+        # lookup the fabrikant corresponding to the current DJ user
+        fabrikant_name = self.user_table.get_current_user()
+        seed = (self.seed_table & key).fetch1("seed")
+
+        # load everything
+        dataloaders, model, trainer = self.load_model(key, include_trainer=True, include_state_dict=True, seed=seed)
+
+        # define callback with pinging
+        def call_back(**kwargs):
+            self.connection.ping()
+            self.call_back(**kwargs)
+
+        trainer_output = trainer(model=model, dataloaders=dataloaders, seed=seed, uid=key, cb=call_back)
```

Comment:
simpler would have been:
```
score, output, model_state, *extra = trainer(...)
transfer_data = extra[0] if extra else {}
```

This avoids the redundant pairing of known fixed outputs (`score, output, model_state`) with hard coded offset of 3 & 4 appearing in the follow up code.